site/pages/create-disruption.page.tsx

```diff
@@ -97,86 +206,99 @@ const CreateDisruption = (initialState: PageState<Partial<DisruptionPageInputs>>
+                            stateUpdater={validityStateUpdater}
                             initialErrors={pageState.errors}
-                            schema={createDisruptionSchema.shape.disruptionEndDate}
+                            reset={!validity.disruptionEndDate}
+                            schema={validitySchema.shape.disruptionEndDate}
                         />
 
-                        <TimeSelector<DisruptionPageInputs>
+                        <TimeSelector<Validity>
                             display="End time"
                             hint="Enter the time in 24hr format. For example 0900 is 9am, 1730 is 5:30pm"
-                            value={pageState.inputs.disruptionEndTime}
-                            disabled={pageState.inputs.disruptionNoEndDateTime === "true"}
+                            value={validity.disruptionEndTime}
+                            disabled={validity.disruptionNoEndDateTime === "true"}
                             inputName="disruptionEndTime"
-                            stateUpdater={stateUpdater}
+                            stateUpdater={validityStateUpdater}
                             initialErrors={pageState.errors}
-                            schema={createDisruptionSchema.shape.disruptionEndTime}
+                            reset={!validity.disruptionEndTime}
+                            schema={validitySchema.shape.disruptionEndTime}
                         />
 
-                        <Checkbox<DisruptionPageInputs>
+                        <Checkbox<Validity>
                             inputName="disruptionNoEndDateTime"
                             display="Does the disruption have an end datetime?"
                             hideLegend
                             checkboxDetail={[
                                 {
                                     display: "No end date/time",
                                     value: "true",
-                                    checked: pageState.inputs.disruptionNoEndDateTime === "true",
-                                },
-                            ]}
-                            stateUpdater={stateUpdater}
-                            initialErrors={pageState.errors}
-                        />
-
-                        <Radios<DisruptionPageInputs>
-                            display="Does this disruption repeat?"
-                            value={pageState.inputs.disruptionRepeats ?? ""}
-                            inputName="disruptionRepeats"
-                            radioDetail={[
-                                {
-                                    value: "yes",
-                                    display: "Yes",
-                                },
-                                {
-                                    value: "no",
-                                    display: "No",
+                                    checked: validity.disruptionNoEndDateTime === "true",
```

Comment:
Update this `checked: !!validity.disruptionNoEndDateTime,`

Reply:
see comment above